protocol_rfcs/managed-commits.md

```diff
@@ -0,0 +1,345 @@
+          commits. In such a scenario, they could backfill required commit themselves (preferably using PUT-if-absent) to unblock themselves.
+
+## Reading managed-commit tables
+
+With `managed-commits` enabled, a table could have some part of table already backfilled and some part of the table yet-to-be-backfilled.
+The precise information about what are the valid un-backfilled commits is maintained by the commit-owner.
+
+E.g.
+```
+_delta_log/00000000000000000000.json
+_delta_log/00000000000000000001.json
+_delta_log/00000000000000000002.json
+_delta_log/00000000000000000002.checkpoint.parquet
+_delta_log/00000000000000000003.json
+_delta_log/00000000000000000003.00000000000000000005.compacted.json
+_delta_log/00000000000000000004.json
+_delta_log/00000000000000000005.json
+_delta_log/00000000000000000006.json
+_delta_log/00000000000000000007.json
+_delta_log/_commits/00000000000000000006.3a0d65cd-4056-49b8-937b-95f9e3ee90e5.json
+_delta_log/_commits/00000000000000000007.016ae953-37a9-438e-8683-9a9a4a79a395.json
+_delta_log/_commits/00000000000000000008.7d17ac10-5cc3-401b-bd1a-9c82dd2ea032.json
+_delta_log/_commits/00000000000000000008.b91807ba-fe18-488c-a15e-c4807dbd2174.json
+_delta_log/_commits/00000000000000000009.41bf693a-f5b9-4478-9434-af7475d5a9f0.json
+_delta_log/_commits/00000000000000000010.0f707846-cd18-4e01-b40e-84ee0ae987b0.json
+_delta_log/_commits/00000000000000000010.7a980438-cb67-4b89-82d2-86f73239b6d6.json
+```
+
+Suppose the commit-owner is tracking:
+```
+{
+  6  -> "00000000000000000006.3a0d65cd-4056-49b8-937b-95f9e3ee90e5.json",
+  7  -> "00000000000000000007.016ae953-37a9-438e-8683-9a9a4a79a395.json",
+  8  -> "00000000000000000008.7d17ac10-5cc3-401b-bd1a-9c82dd2ea032.json",
+  9  -> "00000000000000000009.41bf693a-f5b9-4478-9434-af7475d5a9f0.json"
+}
+```
+
+Delta clients have two choices to read such tables:
+1. Any Delta client can read such table by listing the `_delta_log` directory and reading the delta/checkpoint/log-compaction files.
+   They cannot use the un-backfilled commits in `_delta_log/_commits` directory as they are not aware of the external commit-owner. Because of this, they may see a stale snapshot
+   of the table if the recent commits are not backfilled.
+    - In the above example, such delta implementation will see version 7 as the latest snapshot.
+2. A client that understands managed commits can supplement the file listing result by contacting the commit-owner for information about un-backfilled commits, in order to get the most
+   recent snapshot of the table.
+    - In the above example, a delta implementation could get information about versions 0 through 7 from _delta_log directory and get information about un-backfilled commits (v8, v9) from the commit-owner.
+
+## Maintenance operations on managed-commit tables
+
+[Checkpoints](#checkpoints-1) and [log compaction files](#log-compaction-files) can only be created for commits in the `_delta_log` directory. In other words, in order to
```

Comment:
Should we also add a sentence about this to each of those sections as well? it's kind of an important detail

Reply:
done.